iceoryx_hoofs/test/moduletests/test_objectpool.cpp

```diff
@@ -570,53 +573,64 @@ TEST_F(ObjectPool_test, pointerToIndexConversion)
     // conversion does not care about valid data, index1 still corresponds to ptr1
     EXPECT_THAT(pool.pointerToIndex(ptr1), Eq(index1));
     EXPECT_THAT(pool.indexToPointer(index1), Eq(ptr1));
+}
 
-    // legal pointer checks
-    auto first = reinterpret_cast<Foo*>(pool.m_first);
-    auto last = reinterpret_cast<Foo*>(pool.m_last);
-    auto alignedPtr = reinterpret_cast<Foo*>(pool.m_first + sizeof(Foo));
+TEST_F(ObjectPool_test, pointerToIndexLegalPointerConversion)
+{
+    data = 0;
+    EXPECT_THAT(poolExposed.construct(data), Ne(NO_INDEX));
+
+    auto first = reinterpret_cast<Foo*>(poolExposed.getFirstPtr());
+    auto last = reinterpret_cast<Foo*>(poolExposed.getLastPtr());
+    auto alignedPtr = reinterpret_cast<Foo*>(poolExposed.getFirstPtr() + sizeof(Foo));
 
-    EXPECT_THAT(pool.pointerToIndex(first), Eq(0));
-    EXPECT_THAT(pool.pointerToIndex(last), Eq(CAPACITY - 1));
-    EXPECT_THAT(pool.pointerToIndex(alignedPtr), Eq(1));
+    EXPECT_THAT(poolExposed.pointerToIndex(first), Eq(0));
+    EXPECT_THAT(poolExposed.pointerToIndex(last), Eq(CAPACITY - 1));
+    EXPECT_THAT(poolExposed.pointerToIndex(alignedPtr), Eq(1));
+}
 
+TEST_F(ObjectPool_test, pointerToIndexIllegalPointerConversion)
+{
+    data = 0;
+    EXPECT_THAT(poolExposed.construct(data), Ne(NO_INDEX));
 
-    // illegal pointer checks
-    auto lowPtr = reinterpret_cast<Foo*>(pool.m_first - 1);  // out of reserved memory ranged (too small)
-    auto highPtr = reinterpret_cast<Foo*>(pool.m_first + 1); // out of reserved memory range (too large)
-    // pointer - first is divisable by sizeof(Foo),  assumes Foo is larger than one byte
-    auto nonalignedPtr = reinterpret_cast<Foo*>(pool.m_first + sizeof(Foo) + 1);
+    auto lowOutOfMemoryPtr = reinterpret_cast<Foo*>(poolExposed.getFirstPtr() - 1);
+    auto highOutOfMemoryPtr = reinterpret_cast<Foo*>(poolExposed.getLastPtr() + 1);
 
-    EXPECT_THAT(pool.pointerToIndex(lowPtr), Eq(NO_INDEX));
-    EXPECT_THAT(pool.pointerToIndex(highPtr), Eq(NO_INDEX));
-    EXPECT_THAT(pool.pointerToIndex(nonalignedPtr), Eq(NO_INDEX));
+    int oneByte{1};
```

Comment:
It's nice that you didn't use a magic number. Just another small nitpicking. Could you please use `constexpr size_t ONE_BYTE{1}`. That's nowadays our coding style for constants.

Reply:
nice, done